geopandas/geodataframe.py

```diff
@@ -204,6 +208,8 @@ def __setattr__(self, attr, val):
         # have to special case geometry b/c pandas tries to use as column...
         if attr == "geometry":
             object.__setattr__(self, attr, val)
+            if self._geometry_column_name is None:
+                self._persist_old_default_geometry_colname()
```

Comment:
Here we call this after that the attribute already has been set, so in that case that will never trigger the warning?